app/assets/javascripts/govuk-component/option-select.js

```diff
@@ -0,0 +1,137 @@
+(function ($) {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  function OptionSelect(options){
+    /* This JavaScript provides two fucntional enhancements to option-select components:
+      1) A link that clears all of the checkboxes (referred to here as the "clearing link")
+      2) Open/closing of the list of checkboxes - this is not provided for ie6 and 7 as the performance is too janky.
+    */
+
+    this.$optionSelect = options.el;
+    this.$options = this.$optionSelect.find("input[type='checkbox']");
+
+    // Build clearing link
+    this.$clearingLink = this.attachClearingLink();
+    this.updateClearingLink();
+
+    // Attach event listeners for clearing
+    this.$clearingLink.on('click', this.resetOptions.bind(this));
+    this.$options.on('click', this.updateClearingLink.bind(this));
+
+    // Performance in ie 6/7 is not good enough to support animating the opening/closing
+    // so do not allow option-selects to be collapsible in this case
+    var allowCollapsible = (typeof ieVersion == "undefined" || ieVersion > 7) ? true : false;
+    if(allowCollapsible){
+
+      // Add js-collapsible class to parent for CSS
+      this.$optionSelect.addClass('js-collapsible');
+
+      // Add the little arrow toggle image
+      this.attachOpenCloseToggle();
+
+      // Add open/close listeners
+      this.$optionSelect.find('.container-head').on('click', this.toggleOptionSelect.bind(this));
```

Comment:
OK cool. Yeah (having thought about it) I think you're right. I didn't originally because I read that original bit in the style guide as "if you add a class with javascript, or if your class is only used for javascript stuff then make it js-" but I see now that wasn't quite the right interpretation.